pkgs/by-name/do/dorion/package.nix

```
@@ -1,65 +1,164 @@
 
-  src = fetchurl {
-    url = "https://github.com/SpikeHD/Dorion/releases/download/v${finalAttrs.version }/Dorion_${finalAttrs.version}_amd64.deb";
-    hash = "sha256-cCZikTZ+IU3mq/FkJfeggXLyWIsWG+a2qu1GbgW93WQ=";
+  webkitgtk' = webkitgtk.override { enableExperimental = true; };
+
+  src = fetchFromGitHub {
+    owner = "SpikeHD";
+    repo = "Dorion";
+    rev = "v${version}";
+    hash = "sha256-x0WLDeR0By2hjoMDZvGXVR83+xICHU8YEd5pWf5nkvs=";
+
+    nativeBuildInputs = [ pnpm.configHook ];
   };
 
-  unpackCmd = ''
-    dpkg -X $curSrc .
-  '';
+  shelter = fetchurl {
+    url = "https://raw.githubusercontent.com/uwu/shelter-builds/1bbbab09ccb98574292ba5b658732e13db7b31bd/shelter.js";
+    hash = "sha256-MHb4IwmLmcwjE9LUDsUD3cU1B0LxGTDzpc0lvnWmMfY=";
+    meta = {
+      homepage = "https://github.com/uwu/shelter";
+      sourceProvenance = [ lib.sourceTypes.binaryBytecode ]; # actually, minified JS
+      license = lib.licenses.cc0;
+    };
+  };
 
-  runtimeDependencies = [
-    glib-networking
-    libappindicator
-    libayatana-appindicator
-  ];
+  frontend = stdenv.mkDerivation {
+    pname = "dorion-ui";
+    inherit version src;
+
+    pnpmDeps = pnpm.fetchDeps {
+      inherit pname version src;
+      hash = "sha256-0++DvoDPY2AnLbZ/PMnxMQL60bqbCsWNvlOan7rTm8k=";
+    };
+
+    nativeBuildInputs = [
+      pnpm.configHook
+      nodejs
+    ];
+
+    buildPhase = ''
+      runHook preBuild
+
+      cp ${shelter} src-tauri/injection/shelter.js
+      pnpm build
+
+      runHook postBuild
+    '';
+
+    installPhase = ''
+      runHook preInstall
+
+      mkdir -p $out
+      cp -r src-tauri/{injection,html} $out
+
+      runHook postInstall
+    '';
+  };
+
+in
+
+rustPlatform.buildRustPackage {
+  inherit pname version src;
+
+  sourceRoot = "${src.name}/src-tauri";
+
+  cargoLock = {
+    lockFile = ./Cargo.lock;
+    outputHashes = {
+      "rsrpc-0.13.0" = "sha256-2zRuhtVHIbb1yEot2tT49cl0ESfYFvBX5mWu4HtiH7I=";
+      "tauri-plugin-window-state-0.1.1" = "sha256-Mf2/cnKotd751ZcSHfiSLNe2nxBfo4dMBdoCwQhe7yI=";
+      "window-shadows-0.2.2" = "sha256-qSVDA3SYXi40HSYD9VkfBV1hlduC0xp2kAnXGNtH6UE=";
+      "window_titles-0.1.0" = "sha256-lk2T+6curAwqOUuQ8RtYCjX2ygGBgzt4ILBAMV+ql0w=";
+    };
+  };
 
   nativeBuildInputs = [
+    pkg-config
+    yq-go
     autoPatchelfHook
-    dpkg
     wrapGAppsHook3
   ];
 
   buildInputs = [
-    glib-networking
+    openssl
+    webkitgtk'
     gst_all_1.gst-plugins-bad
     gst_all_1.gst-plugins-base
     gst_all_1.gst-plugins-good
-    webkitgtk
+    glib-networking
+  ];
+
+  runtimeDependencies = [
+    libappindicator
+    libayatana-appindicator
   ];
 
-  installPhase = ''
-    runHook preInstall
+  env = {
+    TAURI_RESOURCE_DIR = "${placeholder "out"}/lib";
+  };
 
-    mkdir -pv $out
-    mv -v {bin,lib,share} $out
+  postPatch = ''
+    # disable pre-build script, change distribution dir from an URL to src, and disable auto-updater
+    rm ../.cargo/config.toml #remove this next update
```

```suggestion
    # disable pre-build script, change distribution dir from an URL to src, and disable auto-updater
    ls -l ../.cargo/config.toml
    rm ../.cargo/config.toml #remove this next update
```

Check the symlink and perms

Are you sure you don't want to remove `.cargo/config.toml` which many packages do?

---

Maybe https://github.com/NixOS/nixpkgs/blob/cf67a1a89dab78f9b425b5a1debe223db4f7819c/pkgs/stdenv/generic/setup.sh#L1315

---

DId it :3